src/integrationTest/resources/application.yml

```diff
@@ -5,11 +5,41 @@ spring:
     platform: h2
     username: ${POSTGRES_USERNAME:dbuserprofile}
     password: ${POSTGRES_PASSWORD:dbuserprofile}
-
+  main:
+    allow-bean-definition-overriding: true
+    security:
+      oauth2:
+        client:
+          provider:
+            oidc:
+              issuer-uri: ${OPEN_ID_API_BASE_URI:https://idam-web-public.aat.platform.hmcts.net/o}
+          registration:
+            oidc:
+              client-id: microservicekey-rd-user-profile-api
+              client-secret: ${S2S_SECRET:K6KA6PRLRFWQLSD6}
 security:
+  anonymousPaths:
+    - "/"
+    - "/csrf"
+    - "/health"
+    - "/health/liveness"
+    - "/loggers/**"
+    - "/swagger-ui.html"
+    - "/swagger-resources/**"
+    - "/v2/api-docs"
+    - "/webjars/springfox-swagger-ui/**"
+    - "/favicon.ico"
+    - "/error"
+
+idam:
+  s2s-auth:
+    totp_secret: ${S2S_SECRET:OLMEJQA7UHU6EZKG}
+    microservice: rd_user_profile_api
+    url: ${S2S_URL:http://127.0.0.1:8990}
 
-  authorisedServices:
-    - "it"
+  s2s-authorised:
+    services: ${PRD_S2S_AUTHORISED_SERVICES:rd_user_profile_api}
+  api:
+    url: ${IDAM_URL:http://127.0.0.1:5000}
 
-auth.idam.client.baseUrl: ${IDAM_URL:http://127.0.0.1:5000}
-auth.provider.service.client.baseUrl: ${S2S_URL:http://127.0.0.1:8099}
\ No newline at end of file
+oidc.issuer: ${OIDC_ISSUER_URL:https://forgerock-am.service.core-compute-idam-aat.internal:8443/openam/oauth2/hmcts}
```

Comment:
will this call actual AAT SIDAM for issuer url for integration test cases? Ideally mocks should be used just like used in PRD just like used in PRD.

Reply:
for user profile not required mock since we are not doing any role based validation with bearer token